orientdb/src/main/scala/akka/stream/alpakka/orientdb/javadsl/OrientDBSource.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.orientdb.javadsl
+
+import akka.NotUsed
+import akka.stream.alpakka.orientdb._
+import akka.stream.javadsl.Source
+import com.orientechnologies.orient.core.record.impl.ODocument
+
+object OrientDBSource {
+
+  /**
+   * Java API: creates a [[OrientDBSourceStage]] that consumes as ODocument
+   */
+  def create(className: String,
+             settings: OrientDBSourceSettings,
+             query: String): Source[OutgoingMessage[ODocument], NotUsed] =
+    Source.fromGraph(
+      new OrientDBSourceStage(
+        className,
+        query,
+        settings,
+        new ODocumentMessageReader[ODocument]()
+      )
+    )
+
+  /**
+   * Java API: creates a [[OrientDBSourceStage]] that consumes as specific type
+   */
+  def typed[T](className: String,
+               settings: OrientDBSourceSettings,
+               clazz: Class[T],
+               query: String = null): Source[OutgoingMessage[T], NotUsed] =
+    Source.fromGraph(
+      new OrientDBSourceStage[T](
+        className,
+        query,
+        settings,
+        new ODocumentMessageReader[T](),
+        clazz = Some(clazz)
+      )
+    )
+
+  private class ODocumentMessageReader[T] extends MessageReader[T] {
+
+    override def convert(oDocs: List[T]): OSQLResponse[T] =
+      try {
+        OSQLResponse(None, Some(OSQLResult(oDocs.map(OutgoingMessage(_)))))
+      } catch {
+        case exception: Exception => OSQLResponse(Some(exception.getMessage), None)
```

Comment:
Maybe `exception.toString` would be more helpful. E.g. NPE has `null` as message.